sources/engine/Xenko.Input/Windows/InputSourceWinforms.cs

```diff
@@ -165,6 +171,20 @@ private static WinFormsKeys GetCorrectExtendedKey(WinFormsKeys virtualKey, long
 
             return virtualKey;
         }
+
+        /// <summary>
+        /// Windows keeps sending KEYDOWN messages while the user holds down the key.
+        /// <br/>This function is used to find out if the received message is a repeated KEYDOWN.
+        /// </summary>
+        /// <param name="lParam">lParam of the KEYDOWN message</param>
+        /// <returns><c>True</c> if this message is a repeated KeyDown, <c>false</c> if it's an actual keydown</returns>
+        private static bool MessageIsDownAutoRepeat(IntPtr lParam)
+        {
+            // According to the microsoft docs on WM_KEYDOWN
+            // (https://docs.microsoft.com/en-us/windows/desktop/inputdev/wm-keydown)
+            // The second to last bit is 0 when the last keyboard message was up and 1 if it was already down
+            return ((int)lParam & (1 << 30)) != 0;
```

Comment:
@xen2 should that be enclosed by `unchecked`? `IntPtr` can be 32-bis or 64-bits depending on the platform if I'm not mistaken.

Reply:
You're right, I'll just use a long since it's already used in the other function.